internal/xds/translator/accesslog.go

```diff
@@ -171,13 +171,38 @@ func buildXdsAccessLog(al *ir.AccessLog, forListener bool) []*accesslog.AccessLo
 	return accessLogs
 }
 
-func convertToKeyValueList(attributes map[string]string) *otlpcommonv1.KeyValueList {
-	if len(attributes) == 0 {
-		return nil
-	}
+// read more here: https://opentelemetry.io/docs/specs/otel/resource/semantic_conventions/k8s/
+const (
+	k8sNamespaceNameKey = "k8s.namespace.name"
+	k8sPodNameKey       = "k8s.pod.name"
+)
 
+func convertToKeyValueList(attributes map[string]string, additionalLabels bool) *otlpcommonv1.KeyValueList {
+	maxLen := len(attributes)
+	if additionalLabels {
+		maxLen += 2
+	}
 	keyValueList := &otlpcommonv1.KeyValueList{
-		Values: make([]*otlpcommonv1.KeyValue, 0, len(attributes)),
+		Values: make([]*otlpcommonv1.KeyValue, 0, maxLen),
+	}
+
+	// always set the k8s namespace and pod name for better UX
```

Comment:
Im unsure where this should live exactly, here or in the gateway-api layer because this is `provider` specific, but this is fine for now

Reply:
address it with TODO.

Reply:
FYI: I'm trying to remove some redunent info in https://github.com/envoyproxy/envoy/pull/28046